cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/CbDelegatedTokenCredentials.java

```diff
@@ -192,16 +192,14 @@ AuthenticationResult acquireNewAccessToken(String resource) throws IOException {
         }
     }
 
-    private AuthenticationResult acquireAccessTokenFromRefreshToken(String resource, String refreshToken, boolean multipleResourceRefreshToken) {
+    private synchronized AuthenticationResult acquireAccessTokenFromRefreshToken(String resource, String refreshToken, boolean multipleResourceRefreshToken) {
```

Comment:
why we need sync block for this?

Reply:
Because we used one of the variables inside in both synchronized and non synchronized context